UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/HealthCheck/AzureBlobStorageHealthCheck.cs

```diff
@@ -0,0 +1,81 @@
+﻿using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using UKHO.ExchangeSetService.Common.Configuration;
+using UKHO.ExchangeSetService.Common.Helpers;
+using UKHO.ExchangeSetService.Common.Logging;
+using UKHO.ExchangeSetService.Common.Storage;
+
+namespace UKHO.ExchangeSetService.Common.HealthCheck
+{
+    public class AzureBlobStorageHealthCheck : IHealthCheck
+    {
+        private readonly IAzureBlobStorageClient azureBlobStorageClient;
+        private readonly ISalesCatalogueStorageService scsStorageService;
+        private readonly IOptions<EssFulfilmentStorageConfiguration> essFulfilmentStorageConfiguration;
+        private readonly ILogger<AzureBlobStorageService> logger;
+
+        public AzureBlobStorageHealthCheck(IAzureBlobStorageClient azureBlobStorageClient,
+                                           ISalesCatalogueStorageService scsStorageService,
+                                           IOptions<EssFulfilmentStorageConfiguration> essFulfilmentStorageConfiguration,
+                                           ILogger<AzureBlobStorageService> logger)
+        {
+            this.azureBlobStorageClient = azureBlobStorageClient;
+            this.scsStorageService = scsStorageService;
+            this.essFulfilmentStorageConfiguration = essFulfilmentStorageConfiguration;
+            this.logger = logger;
+        }
+
+        public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                string[] exchangeSetTypes = essFulfilmentStorageConfiguration.Value.ExchangeSetTypes.Split(",");
+                string storageAccountConnectionString = string.Empty;
+                HealthCheckResult azureBlobStorageHealthStatus = new HealthCheckResult(HealthStatus.Healthy);
+                foreach (string exchangeSetType in exchangeSetTypes)
+                {
+                    var storageAccountWithKey = GetStorageAccountNameAndKey(exchangeSetType);
```

Comment:
Please can we reuse method GetStorageAccountNameAndKey of AzureBlobStorageService.cs instead of creating new method for health check

Reply:
Fixed